vignettes/advs.Rmd

```diff
@@ -0,0 +1,491 @@
+      TRUE ~ NA_character_
+    ),
+    AVISITN = as.numeric(case_when(
+      VISIT == "SCREENING 1" ~ "-1",
+      VISIT == "BASELINE" ~ "0",
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    ))
+  )
+```
+
+## Derive Additional Variables for Anthropometric indicators {#derive_vars}
+
+In order to be able to compare against the reference files we need to
+know each child's age and length/height respectively at the time of each
+vital signs assessment. So these need to be first derived as variables
+in your ADVS.
+
+Remember to ensure that the unit of these matches that of the reference
+files metadata you are comparing against, i.e. "days" for age and "cm"
+for length/height in our examples here.
+
+Note that some of the standard reference files we use here have
+different units and intervals themselves. For example when you look at
+the "by Age" charts WHO provides this by day but CDC gives this every
+half month. So when converting all as days this still leaves the
+challenge of how to join when the patient's current age falls between an
+interval (e.g. between 24.5 and 25 months for CDC). Here in our package
+functions we took the simple approach to use the nearest age from the
+metadata, rather than apply some form of interpolation. An alternative
+method you could use would be to derive current age according to the
+exact same intervals as the metadata in order for an exact join, but in
+our examples we just kept this by days.
+
+### Derived Variables for Parameters by Age
+
+A variable for current analysis age could be achieved using the
+following code (you might first need to derive `BRTHDT` if not already
+available in ADSL):
+
+```{r eval=TRUE}
+# Calculate Current Analysis Age AAGECUR and unit AAGECURU
+advs <- advs %>%
+  derive_vars_duration(
+    new_var = AAGECUR,
+    new_var_unit = AAGECURU,
+    start_date = BRTHDT,
+    end_date = ADT
+  )
+```
+
+Here is how these age variables look:
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  advs,
+  display_vars = exprs(USUBJID, BRTHDT, ADT, AAGECUR, AAGECURU)
+)
+```
+
+### Derived Variables for Weight by Length/Height
+
+Similar to the above, a variable for current analysis length/height
+could be achieved using the following code:
+
+```{r eval=TRUE}
+# Derive Current HEIGHT/LENGTH at each time point Temporary variable
+advs <- advs %>%
+  derive_vars_merged(
+    dataset_add = advs,
+    by_vars = c(get_admiral_option("subject_keys"), exprs(AVISIT)),
+    filter_add = PARAMCD == "HEIGHT" & toupper(VSSTRESU) == "CM",
+    new_vars = exprs(HGTTMP = AVAL, HGTTMPU = VSSTRESU)
+  )
+```
+
+## Derive Additional Parameters for Anthropometric indicators {#derive_params}
+
+Now we get to the most important section where we show how to create the
+new records for each derived pediatrics parameter. Only certain examples
+are included here, but you'll find more parameters in the example
+template script.
```

Comment:
Can you provide the link ?

Reply:
ok, will add `referenced at the bottom of this vignette` rather than a link, as people should always use `use_ad_template()` to get this.